XmlFormatter/src/Formatter/XmlFormatterProvider.cs

```diff
@@ -107,13 +107,42 @@ private async void FormatFile(string inputFilePath, string outputName, SaveOptio
             FireEvent("Loading", "Loading ...");
             XElement fileToConvert = await Task<XElement>.Run(() =>
             {
-                return XElement.Load(inputFilePath);
+                XElement returnElement;
+                try
+                {
+                    returnElement = XElement.Load(inputFilePath);
+                }
+                catch (Exception)
+                {
+                    FireEvent("Input file not valid", "The input file was not valid!");
+                    returnElement = null;
+                }
+                return returnElement;
             });
 
+            if (fileToConvert == null)
+            {
+                return;
+            }
+
             FireEvent("Saving", "Saving ...");
-            await Task.Run(() => fileToConvert.Save(outputName, options));           
+            bool saveSuccess = await Task<bool>.Run(() => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5090949)